eks/simple-with-bastion/main.tf

```diff
@@ -0,0 +1,124 @@
+provider "aws" {
+  region  = var.aws_region
+}
+
+# Get the latest Flatcar Pro AMI available for the given channel
+data "aws_ami" "flatcar_pro_latest" {
+  most_recent = true
+  owners      = ["aws-marketplace"]
+
+  filter {
+    name   = "architecture"
+    values = ["x86_64"]
+  }
+
+  filter {
+    name   = "virtualization-type"
+    values = ["hvm"]
+  }
+
+  filter {
+    name   = "name"
+    values = ["Flatcar-pro-${var.flatcar_channel}-*"]
+  }
+}
+
+# This ssh key can be used to log into the Bastion and from it into the
+# EKS Cluster.
+resource "aws_key_pair" "ssh" {
+  key_name   = var.cluster_name
+  public_key = var.ssh_public_key
+}
+
+resource "aws_security_group" "worker_group_mgmt" {
+  name_prefix = "worker_group_mgmt"
+  vpc_id      = module.vpc.vpc_id
+
+  ingress {
+    from_port = 22
+    to_port   = 22
+    protocol  = "tcp"
+
+    cidr_blocks = [
+      "10.0.0.0/8",
+    ]
+  }
+}
+
+data "aws_availability_zones" "available" {}
+
+module "vpc" {
+  source  = "terraform-aws-modules/vpc/aws"
+
+  name                 = "${var.cluster_name}-vpc"
+  cidr                 = "10.0.0.0/16"
+  azs                  = data.aws_availability_zones.available.names
+  private_subnets      = ["10.0.1.0/24", "10.0.2.0/24", "10.0.3.0/24"]
+  public_subnets       = ["10.0.4.0/24", "10.0.5.0/24", "10.0.6.0/24"]
+  enable_nat_gateway   = true
+  single_nat_gateway   = true
+  enable_dns_hostnames = true
+
+  tags = {
+    "kubernetes.io/cluster/${var.cluster_name}" = "shared"
+  }
+
+  public_subnet_tags = {
+    "kubernetes.io/cluster/${var.cluster_name}" = "shared"
+    "kubernetes.io/role/elb"                    = "1"
+  }
+
+  private_subnet_tags = {
+    "kubernetes.io/cluster/${var.cluster_name}" = "shared"
+    "kubernetes.io/role/internal-elb"           = "1"
+  }
+}
+
+
+module "eks" {
+  source          = "terraform-aws-modules/eks/aws"
+  cluster_name    = var.cluster_name
+  cluster_version = var.cluster_version
+  subnets         = module.vpc.private_subnets
+
+  tags = {
+    Environment = "dev"
+  }
+
+  vpc_id = module.vpc.vpc_id
+
+  workers_additional_policies = [
+    "arn:aws:iam::aws:policy/AmazonS3ReadOnlyAccess"
+  ]
+
+  worker_groups = [
+    {
+      name                          = "workers"
+      ami_id                        = data.aws_ami.flatcar_pro_latest.image_id
+      instance_type                 = var.instance_type
+      root_volume_size              = 20
+      root_volume_type              = "gp2"
+      asg_desired_capacity          = var.worker_group_size
+      additional_security_group_ids = [aws_security_group.worker_group_mgmt.id]
+      key_name                      = aws_key_pair.ssh.key_name
+
+      # Needed with 2605.12.0, can be dropped in the future
```

Comment:
Maybe consider filing an issue about this, otherwise we are bound to forget to remove the workaround after releasing `2605.13.0`. :)

Reply:
Also, does the complex example need the workaround too?

Reply:
Filed https://github.com/kinvolk/PROJECT-flatcar-linux/issues/476 for the cleanup.

It's not necessary for the complex example because that one doesn't use the bootstrap.sh script, which was the one missing the mkdir command.